minio/api.py

```diff
@@ -44,7 +44,8 @@
 
 # Internal imports
 from . import __title__, __version__
-from .compat import urlsplit, range, urlencode, basestring
+from .compat import (urlsplit, queryencode,
```

Comment:
I see issue in your files/code.
It depends what you target for. I see it from the standpoint what I would target for.

I would target for have everything in your lib unicode. I mean all coming and outgoing strings.
For that typically programmer needs to have two things enabled shown as two lines below.

# encoding: utf-8
from __future__ import unicode_literals

First line says that this file is UTF-8.
Second line makes every string  to be unicode string in this file (not just base string with UTF-8 encoding in it).
What you have now is only first line out of two in this file.
So, your file has all base strings with UTF-8. NOT really unicode strings everywhere.
I'm not sure what you want.
You still can try to go with base strings and UTF-8, but that seems assumed to be more complicated way
 with many side effects that needs to be handled.
Hope this can be useful.